k2/csrc/pytorch_context.cu

```diff
@@ -112,12 +114,19 @@ class PytorchCudaContext : public Context {
   torch::Allocator *allocator_;  // NOT owned here
   int32_t gpu_id_;
 };
+#endif
 
 ContextPtr GetCpuContext() { return std::make_shared<PytorchCpuContext>(); }
 
 ContextPtr GetCudaContext(int32_t gpu_id /*= -1*/) {
+#ifdef K2_USE_CUDA
   if (gpu_id < 0) gpu_id = c10::cuda::current_device();
   return std::make_shared<PytorchCudaContext>(gpu_id);
+#else
+  K2_LOG(FATAL) << "\nk2 is compiled without CUDA!"
```

Comment:
How about we print a warning here and return a CpuContext so that we don't need to write those `#if K2_USE_CUDA` in testing code?

Reply:
Fixed.